service/prometheus/scrapeconfig_test.go

```diff
@@ -883,134 +527,17 @@ func Test_Prometheus_GetScrapeConfigs_Deterministic(t *testing.T) {
 // Test_Prometheus_YamlMarshal tests that Prometheus marshals yaml correctly.
 func Test_Prometheus_YamlMarshal(t *testing.T) {
 	tests := []struct {
-		scrapeConfigs []config.ScrapeConfig
+		service v1.Service
```

Comment:
It makes a lot more sense to test that the generated scrape config matches the expected yaml, as opposed to some given one, so I've updated this test.